src/EndpointAndManager.sol

```diff
@@ -38,6 +38,14 @@ abstract contract EndpointAndManager is Endpoint, Manager, Implementation {
         _upgrade(newImplementation);
     }
 
+    /// @notice This function is used to pause the manager and the endpoint
+    function pause() public override onlyOwner {
+        // pause the manager
+        super.pause();
+        // pause the endpoint
+        _pauseEndpoint();
+    }
+
```

Comment:
I've just realised that maybe we don't need this virtual + override behaviour? The inheritance of Pausable will be linearised so here we're basically calling `_pause` twice and the second pause will fail?

Reply:
yea i think it will be linearized. we just need to call one of these functions to pause. The second one won't fail, however. pausing idempotent, so the behavior is the same but yea we don't need to call this twice

Reply:
I think the `whenNotPaused` modifier on `_pause` will cause a revert, no?

Reply:
ah actually we may want to consider removing the modifier on _pause

Reply:
I think this method should only have a single call to `_pause`, which would override the `pause()` functions on both the Endpoint and Manager contracts.